main.tf

```
@@ -7,6 +7,8 @@ data "aws_subnet" "subnet" {
 module "agent_sg" {
   source = "terraform-aws-modules/security-group/aws"
 
+  count = var.create_egress_security_group == true ? 1 : 0
```

does this work instead?

```suggestion
  count = var.create_egress_security_group ? 1 : 0
```